### PR TITLE
CLD-950: Add commands to the logging and objects container

### PR DESCRIPTION
**Pull request by vilius-opendoor**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

* Remove the `secrets` from extraMounts since we are getting rid of the secret. If we don't remove them, the init container for od_secret will not be able to start. Since we're not using secrets from this chart anyway, we can remove them.
* Add the commands that execute the main process, so we can utilize `od_secret`.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

* Add `commands` to logging and objects pod definitions;

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CONTRIBUTING.md) doc
- [ ] I have read the [CLA](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CLA.md)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ spec:
       {{- end }}
       containers:
       - name: splunk-fluentd-k8s-logs
+        command: ["/bin/sh", "-c", "bundle exec fluentd -c /fluentd/etc/fluent.conf"]
```
Comment (xdu-opendoor) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml:64]: can you add a comment on how this is derived? e.g. a link to the dockerfile you looked up.

I also suggest on using comments (if supported) to indicate which parts are overwriten by us

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml [hunk 2]
```diff
@@ -79,10 +80,7 @@ spec:
               fieldRef:
                 fieldPath: metadata.name
           - name: SPLUNK_HEC_TOKEN
-            valueFrom:
-              secretKeyRef:
-                name: {{ template "splunk-kubernetes-logging.secret" . }}
-                key: splunk_hec_token
```
Comment (xdu-opendoor) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml]: this may not be necessary.
would it work if we:
 - keep the `splunk-kubernetes-logging.secret` object and `splunk_hec_token` key
 - set its value to be `od_secret:<vault_path_to_hec_token>` (via helm `values.yaml`, native to the chart without adding more changes to the fork)

`od-secret` is capable of interpreting that early enough and this set up should work.

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml [hunk 3]
```diff
@@ -106,9 +104,6 @@ spec:
         {{- end }}
         - name: conf-configmap
           mountPath: /fluentd/etc
-        - name: secrets
-          mountPath: /fluentd/etc/splunk
-          readOnly: true
```
Comment (xdu-opendoor) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml]: is this mount related to the configmap reference?

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/templates/deployment.yaml [hunk 4]
```diff
@@ -71,10 +72,7 @@ spec:
               fieldRef:
                 fieldPath: spec.nodeName
           - name: SPLUNK_HEC_TOKEN
-            valueFrom:
-              secretKeyRef:
-                name: {{ template "splunk-kubernetes-objects.secret" . }}
-                key: splunk_hec_token
```
Comment (xdu-opendoor) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/templates/deployment.yaml]: same here

**Review by xdu-opendoor — CHANGES_REQUESTED** [review]

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml (hunk 2 above)
Comment (vilius-opendoor) [reply helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml]: Good question. I'll try it in a bit. It will set the env var to the od-secrer

Comment (vilius-opendoor) [reply helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml:85]: btw, so the reason why I did this is so that we can keep the token in 1 place, as opposed to definining it for each chart.

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml (hunk 1 above)
Comment (vilius-opendoor) [reply helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml:64]: If we want to make it obvious, how about we add a changelog file?

Comment (xdu-opendoor) [reply helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml:64]: I slightly prefer adding context in `README.md` and adding something standing out (e.g. `####opendoor override####`) in comments here for quickly identify the content (similar to how git merge conflicts are marked)

File: README.md [hunk 5]
```diff
@@ -1,3 +1,9 @@
+# Opendoor-related CHANGELOG
+
+* 1.5.4-opendoor.1
```
Comment (xdu-opendoor) [inline README.md:3]: I still prefer SEMVER versions

helm charts and its ecosystem (artifactory, CLI tools, etc) expects SEMVER version, and this may break things (e.g. helm chart publish to ECR has a restriction on SEMVER version)

File: docs/index.yaml [hunk 6]
```diff
@@ -2,21 +2,21 @@ apiVersion: v1
 entries:
   splunk-connect-for-kubernetes:
   - apiVersion: v1
-    appVersion: 1.5.4
+    appVersion: 1.5.4-opendoor.1
```
Comment (xdu-opendoor) [inline docs/index.yaml:5]: I'd prefer staying with `1.5.5` or `1.6.0`, samething for subcharts.

this chart is end of support so we don't have to worry about upstream releasing new versions

File: README.md [hunk 7]
```diff
@@ -1,3 +1,9 @@
...
+    * Add a command entries to [helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml](https://github.com/opendoor-labs/splunk-connect-for-kubernetes/blob/master/helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml) and [helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/templates/deployment.yaml](https://github.com/opendoor-labs/splunk-connect-for-kubernetes/blob/master/helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-objects/templates/deployment.yaml)
```
Comment (xdu-opendoor) [inline README.md]: use permanent link. e.g.
https://github.com/opendoor-labs/splunk-connect-for-kubernetes/blob/53d5b0c1e333ad04c16a556aedd137f15c95a630/helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/daemonset.yaml

Comment (xdu-opendoor) [inline README.md]: need context on how did you derived the command
e.g. command value sourced from the official `Dockerfile` (and link the permanent link to the dockerfile line number)

**Review by xdu-opendoor — CHANGES_REQUESTED** [review]

**Review by xdu-opendoor — APPROVED** [review]
